### PR TITLE
figures

### PR DESCRIPTION
**Pull request by jkrumbiegel**
- add `Figure` object which forms the basis of the standard plot commands instead of `Scene`
- Axis becomes the default 2D axis, LScene with the usual 3D axis for 3D (to be replaced / improved later)
- figures allow nested indexing for convenient creation of sublayouts like `scatter(fig[1, 2][3, 4], ...)`
- all layoutables are streamlined via macro and renamed to Axis, Button, Slider, Toggle, Colorbar, Legend, Box (LRect), Label (LText)
- all mutating plot commands return the plot object
- remove old slider and button
- add `content` and `contents` functions to access figure content more easily by figureposition

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Alright, tests may pass now :) I think this is almost ready! We probably should add a few more tests, since in the current state lots of code paths aren't hit, and I already found quite a few bugs in untested code...

File: docs/src/makielayout/layoutables_examples.md [hunk 1]
```diff
@@ -3,9 +3,9 @@ using CairoMakie
 CairoMakie.activate!()
 ```
 
-## LSlider
+## Slider
 
-A simple slider without a label. You can create a label using an `LText` object,
+A simple slider without a label. You can create a label using an `Label` object,
```
Comment (greimel) [inline docs/src/makielayout/layoutables_examples.md:8]:
```suggestion
A simple slider without a label. You can create a label using a `Label` object,
```

File: docs/src/makielayout/layouting.md [hunk 2]
```diff
@@ -37,21 +37,21 @@ main area will be in the scene.
...
-content, like an `LAxis`, the auto-determined width or height will be `nothing`.
+content, like an `Axis`, the auto-determined width or height will be `nothing`.
 
 ### Computed width and height
 
 The computed width and height is the size that the element reports to a `GridLayout`
 that it is a content element of. This can be different from the auto-size if the
 object doesn't want its parent layout to know its auto-size. This is useful if
-you don't want a column to shrink to the size of an `LText`, for example.
+you don't want a column to shrink to the size of an `Label`, for example.
```
Comment (greimel) [inline docs/src/makielayout/layouting.md:49]:
```suggestion
you don't want a column to shrink to the size of a `Label`, for example.
```

File: docs/src/makielayout/tutorial.md [hunk 3]
```diff
@@ -335,7 +335,7 @@ nothing # hide
 Now, we also want to add a color bar for the two heatmaps. Right now, their colors
 are independently scaled from each other. We choose a scale that makes sense for
 both of them (in our case, we know data ranges only from 1 to 3) and assign that
-to both heatmaps. Then we create an `LColorbar` object with one of the heatmaps.
+to both heatmaps. Then we create an `Colorbar` object with one of the heatmaps.
```
Comment (greimel) [inline docs/src/makielayout/tutorial.md:338]:
```suggestion
to both heatmaps. Then we create a `Colorbar` object with one of the heatmaps.
```

File: src/makielayout/layoutables/textbox.jl [hunk 4]
```diff
@@ -347,10 +349,10 @@ function focus!(tb::LTextbox)
 end
 
 """
-    defocus!(tb::LTextbox)
-Defocuses an `LTextbox` so it doesn't receive keyboard input.
+    defocus!(tb::Textbox)
+Defocuses an `Textbox` so it doesn't receive keyboard input.
```
Comment (greimel) [inline src/makielayout/layoutables/textbox.jl:353]:
```suggestion
Defocuses a `Textbox` so it doesn't receive keyboard input.
```

**Review by greimel — COMMENTED** [review]
I suggested four replacements of an -> a.

It's really exciting to see this PR happening!

**Pull request merged**